contrib/terraform/packet/kubespray.tf

```diff
@@ -0,0 +1,62 @@
+# Configure the Packet Provider
+provider "packet" {
+}
+
+resource "packet_ssh_key" "k8s" {
+  count      = "${var.public_key_path != "" ? 1 : 0}"
+  name       = "kubernetes-${var.cluster_name}"
+  public_key = "${chomp(file(var.public_key_path))}"
+}
+
+resource "packet_device" "k8s_master" {
+  depends_on = ["packet_ssh_key.k8s"]
```

Comment:
Is there a reason for this? is the idea that the masters are rebuilt if the keys change?

Reply:
This is to make sure that the SSH key is created before the servers.
Creating the SSH key is much faster than creating the servers, but it's better to be explicit about it.

I've tested that changing things from:
`public_key_path = "~/.ssh/id_rsa"` to `public_key_path = "test_key.pub"` doesn´ *NOT*  redeploy the servers:
```
packet_device.k8s_node[0]: Refreshing state... (ID: 6a775545-e539-4d1f-a2e2-94e799dbda68)
packet_device.k8s_node[1]: Refreshing state... (ID: 2b9ae9fd-62f2-418c-829b-c48c8a1d72cf)
packet_device.k8s_master: Refreshing state... (ID: b9261c19-d763-426a-a35c-2593db83a251)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + packet_ssh_key.k8s
      id:          <computed>
      created:     <computed>
      fingerprint: <computed>
      name:        "kubernetes-t2"
      public_key:  "ssh-rsa [REDACTED]"
      updated:     <computed>


Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

packet_ssh_key.k8s: Creating...
  created:     "" => "<computed>"
  fingerprint: "" => "<computed>"
  name:        "" => "kubernetes-t2"
  public_key:  "" => "ssh-rsa [REDACTED]"
  updated:     "" => "<computed>"
packet_ssh_key.k8s: Creation complete after 1s (ID: 2ffad440-2458-450c-b4f4-94f31c0aff1d)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

So it shouldn't be any bad surprises there.